### ISSUE TITLE
Wallet generates a (spurious?) key when creating a new wallet

### ISSUE THREAD

**Issue by laanwj** [bitcoin/bitcoin#8416]
The wallet code, after creating a new empty wallet, grabs a key from the mempool and gives it the empty label:

```
 cRj4c74AFYdH9W1vbM3UgnaCeGbwztu3qb11NgDReiudtS2Q1ZAr 2016-07-28T10:48:28Z label= # addr=mwR8SR9sfKk7Q8z99xpdcCUNtccq81iRup hdkeypath=m/0'/0'/0'
```

I'm fairly sure this has always been the case, but is this necessary?

Remember that when using wallet encryption, this key will have been generated pre-encryption, and with HD, with the old seed. So if people actually use the address, instead of creating one when they need it, there could be a small risk.

The code to do this is here: https://github.com/bitcoin/bitcoin/blob/master/src/wallet/wallet.cpp#L3306
Reading the comments this appears to be a leftover from the "default address" concept which was removed in 0.4.0? Do we still need it?

**Comment by jonasschnelli** [comment]
Code https://github.com/bitcoin/bitcoin/blob/master/src/wallet/wallet.cpp#L3308

I agree with @laanwj and I don't see a reason for the `defaultKey`.

**Comment by MarcoFalke** [comment]
Meeting comments from https://bitcoincore.org/en/meetings/2016/12/01/#vchdefaultkey-in-wallet:

Sipa would like to get rid of this, however if we do that, a downgrade to an older wallet version would result in failing to rescan.

Given this is not really urgent there’s no need for hacks like dummy keys etc. 0.14 could stop relying on vchDefaultKey, but still write it, and then in 0.15 delete the vchDefaultKey and increase the minimum version to 0.14 so 0.15 wallets will never be openable with 0.13.

**Comment by laanwj** [comment]
This was fixed in #10952. Thanks @achow101.
